java/src/main/kotlin/io/spine/validation/java/generate/option/ValidateFieldGenerator.kt

```diff
@@ -0,0 +1,167 @@
+import io.spine.validation.java.generate.ValidationCodeInjector.ValidateScope.parentPath
+import io.spine.validation.java.generate.ValidationCodeInjector.ValidateScope.violations
+
+/**
+ * The generator for `(validate)` option.
+ *
+ * Generates code for a single field represented by the provided [view].
+ */
+internal class ValidateFieldGenerator(
+    private val view: ValidateField,
+    override val converter: JavaValueConverter
+) : FieldOptionGenerator, EmptyFieldCheck {
+
+    private val field = view.subject
+    private val fieldType = field.type
+    private val getter = message.field(field).getter<Message>()
+    private val declaringType = StringLiteral(field.declaringType.qualifiedName)
+
+    override fun generate(): FieldOptionCode = when {
+        fieldType.isMessage -> validate(getter, fieldType.message.isAny)
+
+        fieldType.isList ->
+            CodeBlock(
+                """
+                for (var element : $getter) {
+                    ${validate(ReadVar("element"), fieldType.list.isAny)}
+                }
+                """.trimIndent()
+            )
+
+        fieldType.isMap ->
+            CodeBlock(
+                """
+                for (var element : $getter.values()) {
+                    ${validate(ReadVar("element"), fieldType.map.valueType.isAny)}
+                }     
+                """.trimIndent()
+            )
+
+        else -> error(
+            "The field type `${fieldType.name}` is not supported by `ValidateFieldGenerator`." +
+                    " Please ensure that the supported field types in this generator match those" +
+                    " used by `ValidatePolicy` when validating the `ValidateFieldDiscovered` event."
+        )
+    }.run { FieldOptionCode(this) }
+
+    /**
+     * Yields an expression to validate the provided [message] if it implements
+     * [io.spine.validate.ValidatableMessage] interface.
+     *
+     * The reported violations are appended to [violations] list, if any.
+     *
+     * If the passed [message] represents [com.google.protobuf.Any], the method will firstly
+     * unpack the enclosed message, and only then validate it.
+     *
+     * Note that not all instances of [com.google.protobuf.Any] can be unpacked:
+     *
+     * 1) Unpacking of the default instance is impossible.
+     * 2) Unpacking of instances with unknown type URLs is also impossible.
+     *
+     * Such instances are always considered valid.
+     *
+     * The default instances of [com.google.protobuf.Message] are considered valid
+     * for singular fields even if the checked message has one or more required fields.
+     * In this case, it is impossible to determine whether the field itself is set
+     * with an invalid message instance or just empty.
+     *
+     * The above statement doesn't apply to repeated and map fields. Within collections,
+     * default instances are considered invalid.
+     *
+     * Details regarding the generated code:
```

Comment:
Honestly, both points are really not easy to comprehend, let alone update after any modifications are made to the code in the future.

I would remove this block altogether.